### PR TITLE
[lz4] Install pc file, cleanup

### PR DESCRIPTION
**Pull request by dg0yt**
- #### What does your PR fix?  
  - [x] Installs the pc file for pkgconfig.
       (Intented to be referenced in libarchive pc file.)
  - [x] Drops unused xxhash dependency.
       (If you don't want the embedded copy, you will need to add more work, in another PR.)
  - [x] Modernizes the portfile
 
- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  all, no

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  yes

### CODE REVIEW

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/lz4/vcpkg.json [hunk 1]
```diff
@@ -1,10 +1,17 @@
 {
   "name": "lz4",
   "version-string": "1.9.3",
```
[the remark this answers is not in the document]
Comment (dg0yt) [reply ports/lz4/vcpkg.json]: Done.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for your help!

File: ports/lz4/portfile.cmake [hunk 2]
```diff
@@ -6,30 +6,35 @@ vcpkg_from_github(
...
+    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/${FILE}"
+        "defined(LZ4_DLL_IMPORT) && (LZ4_DLL_IMPORT==1)"
+        "${DLL_IMPORT}"
+    )
 endforeach()
 
-vcpkg_fixup_cmake_targets()
+vcpkg_cmake_config_fixup()
+vcpkg_fixup_pkgconfig()
+if(EXISTS "${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/lz4.pc")
```
Comment (c72578) [inline ports/lz4/portfile.cmake:34]: The file is called `liblz4.pc`

Comment (dg0yt) [reply ports/lz4/portfile.cmake:34]: Thanks for the pointer. I fixed the filename after the initial submission (cbc79a8), but I missed this occurence. New fix in #20146.

**Pull request merged**